tests/publiccloud/migration.pm

```diff
@@ -0,0 +1,112 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test distribution migration system in public cloud SUSE images
```

Comment:
I think it would make sense to add text from PR description here : 
```
Start a SLES12 SP5 instance

Install distribution migration service package via zypper

Run the migration

Depending on the test call config

Run a standard migration, from SLES12 SP5 to newer version then
verify the instance is running the targeted, migrated version and not
SLES12 SP5
Runs a custom migration and verify the instance is running the
custom product
Runs a migration to a non existing product and verify it did not
migrate, instance at the end of the migration should run SLES12 SP5z
```